### PR TITLE
rubygems and bundler best practices

### PR DESCRIPTION
**Pull request by halfbyte**
Hi,

this PR might be a bit presumptuous, but I started working on it so here it is:

I've noticed that the repo has Gemfile.lock checked in, which is, according to [the gospel](http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/) not recommended. This led me into the proverbial rabbit hole of rspec deprecations in current versions ending in actually removing one test that simply could not be written in a simple way with current rspec (and is actually kinda superflous). 

Please ignore this PR at will if you find this to be too annoying :)

Thanks,

Jan
- Moved development dependencies to gemspec, cleaning up Gemfile
- Removed Gemfile.lock (should not be checked in to git for libs/gems)
- Updated specs to remove rspec deprecation messages
- Removed test in opt_verify_spec.rb that was not working without deprecated rspec methods
  and was implicitly contained in the next spec anyway.

### CODE REVIEW

File: yubikey.gemspec [hunk 1]
```diff
@@ -32,4 +33,10 @@ Gem::Specification.new do |s|
   # OpenSSL is now a dependancy
   s.add_dependency "jruby-openssl" if RUBY_PLATFORM == "java"
   s.add_dependency "openssl" if RUBY_PLATFORM == "ruby"
+  
+  s.add_development_dependency 'rake', ">= 0.8.7"
+  s.add_development_dependency 'rdoc'
+  s.add_development_dependency 'rspec', ">= 2.0"
+  
+
```
Comment (titanous) [inline yubikey.gemspec]: Drop these trailing newlines.

**Comment by titanous** [comment]
LGTM, thanks for the cleanup :)

**Comment by titanous** [comment]
Looks like the Travis build is failing (it was mostly working in the past).

**Comment by halfbyte** [comment]
I'll take a look.

**Comment by halfbyte** [comment]
@titanous Is there a specific reason for the --without-development in the travis config? As my cleanup did away with the test and development groups, this is most likely causing the failure.

**Comment by titanous** [comment]
@halfbyte Not that I know of.

**Comment by halfbyte** [comment]
Doesn't work, seems to be some sort of default. I'm investigating :)

**Comment by halfbyte** [comment]
Ah, wait a sec, there's a rather long backlog, the PR is still in Queue. Will report back.

**Comment by halfbyte** [comment]
While Travis does it's thing I was also looking at the rubinius failures and it looks like the way rbx needs to be configured on travis changed completely. I'll try to get that running as well.

**Comment by titanous** [comment]
Great, thanks for looking into this.

**Comment by halfbyte** [comment]
Rabbit hole indeed :)

Do you think it would be fair game to remove support for 1.8.7? Otherwise we need to specify an older version of rake.

(Totally not sure about the usage and status of 1.8.7, haven't used it in ages, but people still do, I guess.)

**Comment by titanous** [comment]
Go for it, I've already removed 1.8 support from the other gems I maintain.

**Comment by halfbyte** [comment]
There you go. I kind of dislike that extra block in the Gemfile, but according to [this](https://github.com/travis-ci/travis-ci/issues/1641) this seems to be the way to go.

This was fun :)

**Comment by titanous** [comment]
Awesome, thanks!

**Pull request merged**